typing/ctype.ml

```diff
@@ -3262,13 +3266,10 @@ let filter_self_method env lab priv meths ty =
 *)
 let moregen_occur env level ty =
   let rec occur ty =
-    mark_type_node ty ~guard:
-      begin fun ty ->
-        ty.level > level &&
-        if is_Tvar ty && ty.level >= generic_level - 1 then raise Occur
-        else true
-      end
-      ~after:(iter_type_expr occur)
+    let ty = repr ty in
+    if ty.level <= level then () else
```

Comment:
I think this line is a bit subtle since it deliberately includes marked nodes alongside nodes that have a level less than `level`. It would probably be clearer if this code did `if not_marked_node ty ...` and `flip_mark_node` instead of using `try_mark_node`.

Reply:
Actually, this code mimics the original version, which only compared with `level`.
The current version is more robust, since we use `try_mark_node` before recursing.
The only assumption in the code is that marking a node will not set its level to `generic_level - 1`, which seems reasonable to me.
Note that separately I have removed `mirror_level` and `pivot_level` from the interface of `Btype`, which led to rewrite `Ctype.find_lowest_level` using `not_marked_node/flip_mark_node`.